### PR TITLE
Make `PluginViewRegistry` class more extensible

### PR DESCRIPTION
**Pull request by EstherPerelman**
Signed-off-by: Esther Perelman <esther.perelman@sap.com>

#### What it does
Move some code to separate functions to make it easy to override some logic, This should help with the fix of #9800  [as suggested here](https://github.com/eclipse-theia/theia/issues/9800#issuecomment-891789885)

#### How to test

- `f1` -> `View: Reset Layout workbench` -> see the icons on the left sidebar initialized well.
- `f1` -> `toggle <some_view> view` -> toggle view as expected.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1:

**Review by msujew — APPROVED** [review]
The changes look reasonable to me as well.

**Comment by EstherPerelman** [comment]
@vince-fugnitto @msujew thank you for approving! are you able to merge?

File: packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts [hunk 1]
```diff
@@ -240,17 +252,7 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
...
-                if (widget) {
-                    widget.dispose();
-                } else {
-                    widget = await this.openViewContainer(id);
-                    if (widget) {
-                        this.shell.activateWidget(widget.id);
-                    }
-                }
-            }
+            execute: async () => { await this.toggleViewContainer(id); }
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts:255]: I thought I had already commented on this line but we can do the following to simplify:

```suggestion
            execute: () => this.toggleViewContainer(id);
```

Comment (EstherPerelman) [reply packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts]: Right, I thought of it but saw [this](https://github.com/eclipse-theia/theia/blob/625c021715f0e9d1c91015049656a7d1225a6511/packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts#L317) ...
Fixed.

**Comment by EstherPerelman** [comment]
@vince-fugnitto @msujew @colin-grant-work 
I need this fix... I'll be happy to see it merged...

**Pull request merged**